src/Converter.hh

```diff
@@ -36,10 +37,14 @@ namespace sdf
   {
     /// \brief Convert SDF to the specified version.
     /// \param[in] _doc SDF xml doc
-    /// \param[in] _toVersion Version number in string format
-    /// \param[in] _quiet False to be more verbose
+    /// \param[in] _toVersion Version number in string format.
+    /// \param[out] _errors Vector of errors.
```

Comment:
it feels a little weird to put an output variable in the middle of the argument list. I think I favor either making output variables the first argument (like [SemanticPose::Resolve](https://github.com/gazebosim/sdformat/blob/sdformat12_12.5.0/include/sdf/SemanticPose.hh#L73-L74)) so that subsequent input parameters could be optional.

Also, we sometimes pass `Errors` by reference and sometimes return from the function. It feels weird to both return `bool` and write to an `Errors` vector though.

@azeey what do you think?

Reply:
> it feels a little weird to put an output variable in the middle of the argument list. I think I favor either making output variables the first argument (like [SemanticPose::Resolve](https://github.com/gazebosim/sdformat/blob/sdformat12_12.5.0/include/sdf/SemanticPose.hh#L73-L74)) so that subsequent input parameters could be optional.

Sounds good to me, will apply the changes on this PR.

> Also, we sometimes pass `Errors` by reference and sometimes return from the function. It feels weird to both return `bool` and write to an `Errors` vector though.

This seems a bit more tricky since not all the time an error is added to the structure `false` is returned, meaning `errors.empty() != return false`, so I'm not sure if there would be trivial way of checking this without changing behavior.

Reply:
Just realized, shall we make the `ParserConfig` also optional?

Reply:
Since this is an internal function, I recommend against making `ParserConfig` optional. Whoever is calling should be passing a config.